refactoring/README.md

```diff
@@ -0,0 +1,15 @@
+# Refactoring
+
+Refactoring is very important in relation to these topics. Just as important as the other topics covered here, is how to take good code and turn it into great code.
+
+We can use [design patterns](patterns/README.md) to DRY up code and generalize abstractions. We must avoid [anti-patterns](anti_patterns/README.md) while we do this. While they may be tempting to employ, their costs outweigh their benefits.
```

Comment:
We should link "DRY" for "Don't repeat yourself" wiki page.

Reply:
definitely!